gframe/duelclient.cpp

```diff
@@ -334,8 +337,42 @@ void DuelClient::HandleSTOCPacketLan(char* data, unsigned int len) {
 		str.append(msgbuf);
 		myswprintf(msgbuf, L"%ls%d\n", dataManager.GetSysString(1233), pkt->info.draw_count);
 		str.append(msgbuf);
-		if(pkt->info.duel_rule != DEFAULT_DUEL_RULE) {
-			myswprintf(msgbuf, L"*%ls\n", dataManager.GetSysString(1260 + pkt->info.duel_rule - 1));
+		uint32 rule = 5;
+		mainGame->dInfo.duel_rule = 2;
+		if(pkt->info.duel_flag == MASTER_RULE_1) {
+			rule = 1;
+			mainGame->dInfo.duel_rule = 1;
+		}
+		else if(pkt->info.duel_flag == MASTER_RULE_2) {
+			rule = 2;
+			mainGame->dInfo.duel_rule = 2;
+		}
+		else if(pkt->info.duel_flag == MASTER_RULE_3) {
+			rule = 3;
+			mainGame->dInfo.duel_rule = 3;
+		}
+		else if(pkt->info.duel_flag == MASTER_RULE_4) {
```

Comment:
Official is using **New Master Rule**.
http://www.yugioh-card.com/japan/howto/new_master_rule/

Reply:
None of the Konami rule sets use a consistent naming convention so refering to them by the English colloquialism is fine. There have been 4 rule sets.